src/components/Dashboard/ThisWeek/NewsletterMenu/MenuItems/LogNewsLetter/LogNewsletter.test.tsx

```diff
@@ -1,15 +1,160 @@
+      await waitFor(() => expect(getByText('Save')).not.toBeDisabled());
+      userEvent.click(getByText('Save'));
+      await waitFor(() => expect(handleClose).toHaveBeenCalled());
+    });
+
+    it('Logs Email Newsletter', async () => {
+      const { getByRole, getByText, findByText } = render(
+        <MuiPickersUtilsProvider utils={LuxonUtils}>
+          <AppProvider
+            initialState={{
+              user: {
+                id: 'user-1',
+                firstName: 'John',
+                lastName: 'Smith',
+              } as User,
+            }}
+          >
+            <SnackbarProvider>
+              <GqlMockedProvider<CreateTaskMutation>>
+                <LogNewsletter
+                  accountListId={accountListId}
+                  handleClose={handleClose}
+                />
+              </GqlMockedProvider>
+            </SnackbarProvider>
+          </AppProvider>
+        </MuiPickersUtilsProvider>,
+      );
+      expect(getByText('Log Newsletter')).toBeInTheDocument();
+
+      userEvent.click(getByText('Save'));
+      expect(await findByText('Field is required')).toBeInTheDocument();
+
+      userEvent.type(getByRole('textbox', { name: 'Subject' }), accountListId);
+      await waitFor(() => expect(getByText('Save')).not.toBeDisabled());
+      userEvent.click(getByRole('radio', { name: 'Newsletter - Email' }));
+      userEvent.click(getByText('Save'));
+      await waitFor(() => expect(handleClose).toHaveBeenCalled());
+    });
+
+    it('Logs Newsletter with Comment', async () => {
+      const { getByRole, getByText, findByText } = render(
+        <MuiPickersUtilsProvider utils={LuxonUtils}>
+          <AppProvider
+            initialState={{
+              user: {
+                id: 'user-1',
+                firstName: 'John',
+                lastName: 'Smith',
+              } as User,
+            }}
+          >
+            <SnackbarProvider>
+              <GqlMockedProvider<
+                CreateTaskMutation & CreateTaskCommentMutation
+              >>
+                <LogNewsletter
+                  accountListId={accountListId}
+                  handleClose={handleClose}
+                />
+              </GqlMockedProvider>
+            </SnackbarProvider>
+          </AppProvider>
+        </MuiPickersUtilsProvider>,
+      );
+      expect(getByText('Log Newsletter')).toBeInTheDocument();
+
+      userEvent.click(getByText('Save'));
+      expect(await findByText('Field is required')).toBeInTheDocument();
+
+      userEvent.type(getByRole('textbox', { name: 'Subject' }), accountListId);
+      await waitFor(() => expect(getByText('Save')).not.toBeDisabled());
+      userEvent.click(getByRole('radio', { name: 'Newsletter - Email' }));
+      userEvent.type(
+        getByRole('textbox', { name: 'Comment' }),
+        'Some Random Comment',
+      );
+
+      // userEvent.click(getByText('Save'));
+
+      // await waitFor(() => expect(handleClose).toHaveBeenCalled());
```

Comment:
For some reason, I can't get this test to pass. I keep getting an error about ```cannot read property 'createevent' of null```. Let me know if you guys have any suggestions. It does have something to do with the logic for creating a comment but I'm not sure how to fix it.

Reply:
See if there are any GraphQL errors thrown.  What file/callstack is throwing that `createevent` one?

Reply:
This is all I get in the console: 
<img width="1830" alt="Screen Shot 2021-04-29 at 3 44 23 PM" src="https://user-images.githubusercontent.com/39680460/116608999-d8113580-a901-11eb-8f9f-1809b5e3f3b2.png">

I tried using that ```jdb``` node debugger you showed me but I'm getting some sort of error trying to connect to that as well. Is there another way to debug?

Reply:
@OzzieOrca After playing around with it a bit I ended up getting a graphql error. Does this mean the ```createdAt``` field isn't getting mocked properly? Maybe related to @gyasistory's issue?
<img width="1884" alt="Screen Shot 2021-04-30 at 11 12 47 AM" src="https://user-images.githubusercontent.com/39680460/116717067-7bb42180-a9a6-11eb-9394-0f765d9d4f7e.png">

Reply:
Ya that sounds like the same issue. Let me know if you need help looking into it. https://github.com/SurveyMonkey/graphql-ergonomock#providing-default-mock-resolver-functions might be helpful.